replica/distributed_loader.cc

```diff
@@ -155,9 +155,8 @@ collect_all_shared_sstables(sharded<sstables::sstable_directory>& dir, sharded<r
         auto shared_sstables = d.retrieve_shared_sstables();
         sstables::sstable_directory::sstable_open_info_vector need_cleanup;
         if (sorted_owned_ranges_ptr) {
-            auto& table = db.local().find_column_family(ks_name, table_name);
             co_await d.filter_sstables([&] (sstables::shared_sstable sst) -> future<bool> {
-                if (table.update_sstable_cleanup_state(sst, *sorted_owned_ranges_ptr)) {
+                if (needs_cleanup(sst, *sorted_owned_ranges_ptr)) {
```

Comment:
BTW, the reason this wasn't the root cause for #14001 is that here we filter the unshared sstables, meaning it's benign to cleanup those sstables later on from the multi-shard sharing perspective.
But the hunk is still needed.

The original intent was to detect sstables requiring cleanup on start-up, when populating the table.
But we're not there yet as we don't have the token metadata in this early stage.